src/ext_vec.rs

```diff
@@ -425,15 +425,10 @@ pub trait ByteVec: Sealed {
     where
         Self: Sized,
     {
-        let v = self.as_vec();
-        if let Ok(allutf8) = v.to_str() {
-            return allutf8.to_string();
+        match self.as_vec().to_str_lossy() {
+            Cow::Borrowed(_) => unsafe { self.into_string_unchecked() },
```

Comment:
This LGTM, but could you add a safety note here? Like this:

```rust
// SAFETY: blah blah
```

I believe every other use of `unsafe` in this crate has a safety annotation. Also, I try not to use the word `unsafe` in the description so that grep reports fewer false positives.

Thanks!

Reply:
Thanks for the quick review. Added the comment. :)